modules/core/shared/src/main/scala-3/io/circe/Derivation.scala

```diff
@@ -1,300 +1,52 @@
 package io.circe
 
-import cats.data.{ NonEmptyList, Validated }
+import scala.compiletime.constValue
 import scala.deriving.Mirror
-import scala.collection.mutable.WrappedArray
-import scala.compiletime.{ constValue, erasedValue, error, summonFrom }
+import Predef.genericArrayOps
+import cats.data.{ NonEmptyList, Validated }
+import io.circe.derivation._
 
+@deprecated(since = "0.14.4")
 object Derivation {
-
   inline final def summonLabels[T <: Tuple]: Array[String] = summonLabelsRec[T].toArray
-  inline final def summonDecoders[T <: Tuple]: Array[Decoder[_]] = summonDecodersRec[T].toArray
-  inline final def summonEncoders[T <: Tuple]: Array[Encoder[_]] = summonEncodersRec[T].toArray
-
-  inline final def summonEncoder[A]: Encoder[A] = summonFrom {
-    case encodeA: Encoder[A] => encodeA
-    case _: Mirror.Of[A]     => Encoder.AsObject.derived[A]
-  }
-
-  inline final def summonDecoder[A]: Decoder[A] = summonFrom {
-    case decodeA: Decoder[A] => decodeA
-    case _: Mirror.Of[A]     => Decoder.derived[A]
-  }
+  inline final def summonDecoders[T <: Tuple]: Array[Decoder[_]] = summonDecoders[T].toArray
+  inline final def summonEncoders[T <: Tuple]: Array[Encoder[_]] = summonEncoders[T].toArray
```

Comment:
These are accidentally infinitely recursive at runtime 
```suggestion
  inline final def summonDecoders[T <: Tuple]: Array[Decoder[_]] = derivation.summonDecoders[T](using Configuration.default).toArray
  inline final def summonEncoders[T <: Tuple]: Array[Encoder[_]] = derivation.summonEncoders[T](using Configuration.default).toArray
```